src/features/HomePage/index.tsx

```diff
@@ -1,61 +1,238 @@
 import { useNavigate } from 'react-router-dom';
-import { useTranslation } from 'react-i18next';
 import styled from 'styled-components';
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
 
+import { fetchNotices } from './noticesApi';
+import {
+  Mentor,
+  selectFilteredMentors,
+  useGetMentorsQuery,
+} from '../MentorPage/mentorPageApi';
 import { selectHasUnreadMessages } from '@/features/Chat/chatSlice';
 import { useAppSelector } from '@/store';
+import { useGetLayoutMode } from '@/hooks/useGetLayoutMode';
 
+import Background from '@/static/img/mountain-background.svg';
+import ListCard from '../MentorPage/components/MentorList/MentorCard/List';
+import MentorCard from '../MentorPage/components/MentorList/MentorCard/Expanded';
 import NewMessagesImage from '@/static/img/new-messages.svg';
 import PageWithTransition from '@/components/PageWithTransition';
-import { palette } from '@/components/variables';
+import {
+  NAVIGATION_HEIGHT,
+  OUTER_HORIZONTAL_MARGIN,
+  palette,
+} from '@/components/variables';
 import Text from '@/components/Text';
 import { TextButton } from '@/components/Buttons';
+import { UserRole, selectUserRole } from '../Authentication/userSlice';
+
+import type { Notice } from './noticesApi';
 
 const HomePage = () => {
   const navigate = useNavigate();
   const { t } = useTranslation('home');
 
   const unreadMessagesFound: boolean = useAppSelector(selectHasUnreadMessages);
+  const navigateToAdminPanel = () => navigate('/admin');
   const navigateToChat = () => navigate('/chat');
+  const navigateToMentors = () => navigate('/mentors');
+
+  const userRole: UserRole | null = useAppSelector(selectUserRole);
```

Comment:
I dont think we need to explicitly add the return type here. 
If it is needed, I would rather add them to the selector implementation itself. Also for line 18 and 48

Reply:
You're right, they were just there to clarify the variable types for myself. Now removed. 👍